c7n/resources/vpc.py

```diff
@@ -1600,3 +1600,79 @@ class resource_type(object):
         name = 'KeyName'
         date = None
         dimension = None
+
+
+@Vpc.action_registry.register('set-vpc-flow')
+@Subnet.action_registry.register('set-vpc-flow')
+@NetworkInterface.action_registry.register('set-vpc-flow')
+class CreateFlowLogs(BaseAction):
+    """Create flow logs for a network resource
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: vpc-enable-flow-logs
+            resource: vpc
+            filters:
+              - type: flow-logs
+                enabled: false
+            actions:
+              - type: set-vpc-flow
+                DeliverLogsPermissionArn: arn:iam:role
+                LogGroupName: /custodian/vpc/flowlogs/
+    """
+    permissions = ('ec2:CreateFlowLogs',)
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-vpc-flow']},
+            'DeliverLogsPermissionArn': {'type': 'string'},
+            'LogGroupName': {'type': 'string'},
+            'TrafficType': {'type': 'string',
+                            'enum': ['ACCEPT', 'REJECT', 'ALL']},
+            'required': ['LogGroupName']
+        }
+    }
+
+    def validate(self):
+        if not self.data.get('DeliverLogsPermissionArn'):
+            if self.data.get('LogGroupName'):
+                raise ValueError('DeliverLogsPermissionArn required when '
+                                 'specifying LogGroupName')
+        return self
+
+    def delete_flow_logs(self, client, rids):
+        flow_logs = client.describe_flow_logs(
+            Filters=[{'Name': 'resource-id', 'Values': rids}])['FlowLogs']
+        client.delete_flow_logs(FlowLogIds=[f['FlowLogId'] for f in flow_logs])
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        params = dict(self.data)
+        params.pop('type')
+
+        model = self.manager.get_model()
+        r_type = model.type
+        if r_type == 'vpc':
+            r_type = 'VPC'
+        elif r_type == 'subnet':
+            r_type = 'Subnet'
+        else:
+            r_type = 'NetworkInterface'
+
+        params['ResourceType'] = r_type
+        params['TrafficType'] = self.data.get('TrafficType', 'ALL').upper()
+        params['ResourceIds'] = [r[model.id] for r in resources]
+
+        if not params['ResourceIds']:
```

Comment:
this conditional can be elided/removed, policies execution won't invoke an action with an empty set of resources.